CS3219-AY2223S1/cs3219-project-ay2223s1-g58
# Add dark mode support

**What is the purpose of this pull request?**


**Overview of changes**
Fixes #106 

- Add dark mode toggle and it's mechanism through chakra-ui + tailwindcss
	- chakra UI components will typically have compatible dark mode color
	- for tailwind CSS, a `dark:xxxx` class can be added to set style when dark mode is on https://tailwindcss.com/docs/dark-mode
- Frontend fixes to ensure text and background colors update accordingly
- Update a way to run just the frontend on development and have the rest of the services ready
- Add two hooks currently not in use but just in case

**How it was tested/How to test**
- Run all services according to root readme
- See that the dark mode can be toggled and the UI should still be readable/reasonable

**Service(s) Touched**
